cmd/eval.go

```diff
@@ -393,7 +400,11 @@ func eval(args []string, params evalCommandParams, w io.Writer) (bool, error) {
 
 	if err != nil {
 		return false, err
-	} else if len(result.Errors) > 0 {
+	} else if errorCount := len(result.Errors); errorCount > 0 && errorCount != builtInErrorCount {
+		// if we only have built-in errors, we don't want to return an error. If
+		// strict-builtin-errors is set the first built-in error will be returned
+		// in a result error instead.
+
```

Comment:
[comment] This is an interesting use of the "assignment in an if-stmt" construct in Go. Fits perfect for the use case. :+1: 

I also appreciate the comment explaining _why_ the above conditionals are the way they are-- I'll admit that it was not instantly obvious to me when I first glanced at that line. :smile: